store/src/java/com/zimbra/cs/db/DbSearch.java

```diff
@@ -173,6 +175,10 @@ static String orderBy(SortBy sort, boolean alias) {
         if (sort.getKey() == SortBy.Key.NONE) { // no ORDER BY for NONE
             return "";
         }
+        if (sort.getKey() == SortBy.Key.UNREAD) {
+            return " ORDER BY " + (alias ? SORT_COLUMN_ALIAS : toSortField(sort))
+                + (sort.getDirection() == SortBy.Direction.DESC ? " DESC" : "") + ", mi.id DESC";
```

Comment:
Why mi.id DESC by default?  Interestingly, this is very close to a change I propose in https://github.com/Zimbra/zm-mailbox/pull/513.  Maybe I should change that to default to DESC or perhaps use the same direction as the first sort.  Would be interested in your thoughts.

Reply:
to keep chronological order(latest on top) always for group of read and unread mails.
So, first sort by unread flag(Asc/Desc) and sort by id(Desc)

Reply:
Note that I've committed my changes, so you will need to rebase - sorry.